backend/functions/src/course/functions.ts

```diff
@@ -1,8 +1,23 @@
 import { db } from '../config'
 import { getStudentsData } from '../student/functions'
 import admin from 'firebase-admin'
-import { Course, FirestoreCourse, FirestoreGroup } from '../types'
+import { Course, FirestoreCourse, FirestoreGroup, Semester } from '../types'
 const courseRef = db.collection('courses')
+const semesterRef = db.collection('utils').doc('semester')
+
+export const getCurrentSemester = async (): Promise<String> => {
+  const semData = (await semesterRef.get()).data() as Semester
+  return semData.currentSemester
+}
+
+export const setCurrentSemester = async (sem: String) => {
```

Comment:
For `setCurrentSemester`, I think it may be necessary to check, and add the semester to `allSemesters` just in case we are adding a new semester. Apart from that the endpoints looks good!